### PR TITLE
Upgrade to Scala 2.13

### PR DESCRIPTION
**Pull request by mario-galic**
## What does this change?

- Migrate to Scala 2.13
- The following guardian libraries had to be released for 2.13
  - content-authorisation-common https://github.com/guardian/content-authorisation-common/pull/6
  - support-internationalisation https://github.com/guardian/support-frontend/pull/2676

## Have we considered potential risks?

- Moving from support-internationalisation 0.9 to 0.13 means Turkey is classified as ROW instead of Domestic as per https://github.com/guardian/support-libraries/pull/13
- Upgrade is made riskier by the fact that there are many libraries (some performing duplicated functionality) in `Dependencies.scala`.

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -4,51 +4,69 @@ import sbtassembly.AssemblyPlugin.autoImport.{MergeStrategy, assemblyMergeStrate
...
   val awsStepFunction = "com.amazonaws" % "aws-java-sdk-stepfunctions" % awsVersion
   val awsSdkLambda = "com.amazonaws" % "aws-java-sdk-lambda" % awsVersion
   val awsLambda = "com.amazonaws" % "aws-lambda-java-core" % "1.2.0"
-  val supportInternationalisation = "com.gu" %% "support-internationalisation" % "0.9"
+  val scalaLambda = "io.github.mkotsur" %% "aws-lambda-scala" % "0.2.0"
+
+  // Cats
+  val catsCore = "org.typelevel" %% "cats-core" % catsVersion
+  val catsEffect = "org.typelevel" %% "cats-effect" % catsEffectVersion
+  val mouse = "org.typelevel" %% "mouse" % "0.23" // can be removed once we move to Scala 2.13 (native 'tap')
```
Comment (kelvin-chappell) [inline project/Dependencies.scala:31]:
```suggestion
   
```

**Review by kelvin-chappell — APPROVED** [review]
Looks good to me 👍

File: handlers/delivery-problem-credit-processor/src/main/scala/com/gu/deliveryproblemcreditprocessor/DeliveryCreditProcessor.scala [hunk 2]
```diff
@@ -24,6 +24,7 @@ import zio.{Task, ZIO}
 
 object DeliveryCreditProcessor extends Logging {
 
+  implicit val cs: ContextShift[IO] = IO.contextShift(scala.concurrent.ExecutionContext.global)
```
Comment (mario-galic) [inline handlers/delivery-problem-credit-processor/src/main/scala/com/gu/deliveryproblemcreditprocessor/DeliveryCreditProcessor.scala:27]: The reason for these ZIO changes are a result of me trying to bump ZIO version. I have since reverted the version upgrade to do it in a separate PR, but I have left these changes in preparation for ZIO version upgrade.

File: project/Dependencies.scala (hunk 1 above)
Comment (mario-galic) [reply project/Dependencies.scala:31]: I will do this in separate PR.

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,6 +1,5 @@
-addSbtPlugin("com.gu" % "sbt-riffraff-artifact" % "1.1.4")
-
 resolvers += Resolver.typesafeRepo("releases")
+addSbtPlugin("com.gu" % "sbt-riffraff-artifact" % "1.1.4")
 addSbtPlugin("org.scalariform" % "sbt-scalariform" % "1.8.2")
+addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.6")
 
-addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.10.0-RC1")
```
Comment (johnduffell) [inline project/plugins.sbt:6]: @mario-galic this plugin is required for `snyk monitor` to work in the build, now it is (silently) not running any more.
There probably should have been a comment on this line explaining that though.

```
[11:06:12] :		 [Step 1/1] >>> command: sbt "-Dsbt.log.noformat=true" dependencyTree >>> exit code: 1 >>> stdout: [info] welcome to sbt 1.3.13 (Private Build Java 1.8.0_252)
[11:06:12] :		 [Step 1/1] [info] loading settings for project root-3cc2acc8e92de785-build from plugins.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading project definition from /opt/teamcity/buildAgent/work/3cc2acc8e92de785/project
[11:06:12] :		 [Step 1/1] [info] loading settings for project contact-us-api from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project digital-voucher-suspension-processor from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project digital-voucher-cancellation-processor from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project digital-voucher-api from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project delivery-records-api from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project fulfilment-date-calculator from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project sf-move-subscriptions-api from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project metric-push-api from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project delivery-problem-credit-processor from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project holiday-stop-processor from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project braze-to-salesforce-file-upload from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project batch-email-sender from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project zuora-datalake-export from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project sf-datalake-export from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project holiday-stop-api from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project sf-gocardless-sync from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project cancellation-sf-cases-api from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project sf-billing-account-remover from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project sf-contact-merge from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project dev-env-cleaner from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project zuora-sar from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project zuora-retention from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project new-product-api from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project identity-retention from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project catalog-service from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project digital-subscription-expiry from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project identity-backfill from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project zuora-callout-apis from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] loading settings for project root-3cc2acc8e92de785 from build.sbt ...
[11:06:12] :		 [Step 1/1] [info] resolving key references (39262 settings) ...
[11:06:12] :		 [Step 1/1] [info] set current project to root-3cc2acc8e92de785 (in build file:/opt/teamcity/buildAgent/work/3cc2acc8e92de785/)
[11:06:12] :		 [Step 1/1] [error] Expected ';'
[11:06:12] :		 [Step 1/1] [error] Not a valid command: dependencyTree
[11:06:12] :		 [Step 1/1] [error] Not a valid project ID: dependencyTree
[11:06:12] :		 [Step 1/1] [error] Expected ':'
[11:06:12] :		 [Step 1/1] [error] Not a valid key: dependencyTree (similar: dependencyOverrides, sbtDependency, dependencyResolution)
[11:06:12] :		 [Step 1/1] [error] dependencyTree
[11:06:12] :		 [Step 1/1] [error]               ^
[11:06:12] :		 [Step 1/1]  null
[11:06:12] :		 [Step 1/1] 
[11:06:12] :		 [Step 1/1] Please make sure that the `sbt-dependency-graph` plugin (https://github.com/jrudolph/sbt-dependency-graph) is installed globally or on the current project, and that `sbt "-Dsbt.log.noformat=true" dependencyTree` executes successfully on this project.
[11:06:12] :		 [Step 1/1] 
[11:06:12] :		 [Step 1/1] Alternatively you can use `sbt-coursier` for dependency resolution (https://get-coursier.io/docs/sbt-coursier), in which case ensure that the plugin is installed on the current project and that `sbt "-Dsbt.log.noformat=true" coursierDependencyTree` executes successfully on this project.
[11:06:12] :		 [Step 1/1] 
[11:06:12] :		 [Step 1/1] For this project we guessed that you are using sbt-dependency-graph.
```

cc @ripecosta

Comment (mario-galic) [reply project/plugins.sbt:6]: This plugin has been absorbed by sbt 1.4.0. I'll open a PR.

Comment (mario-galic) [reply project/plugins.sbt:6]: Resolved by https://github.com/guardian/support-service-lambdas/pull/732
